src/repart/repart.c

```diff
@@ -7487,7 +7510,41 @@ static int context_fstab(Context *context) {
                 }
         }
 
-        r = flink_tmpfile(f, t, path, 0);
+        switch (arg_append_fstab) {
+        case APPEND_AUTO:
+                fputs(AUTOMATIC_FSTAB_HEADER_END "\n", f);
+
+                r = read_full_file(path, &c, &l);
+                if (r < 0) {
+                        if (r == -ENOENT) {
+                                log_warning("File fstab not found in %s", path);
+                                break;
+                        }
+                        return log_error_errno(r, "Failed to open %s: %m", path);
+                }
+
+                if (startswith(c, "# Automatically generated by systemd-repart\n\n")) {
+                        log_warning("The fstab was generated by systemd-repart and will not be included in the new one. Ignoring.");
+                        break;
+                }
+                char *acs;
+                const char *ace;
+                acs = find_line(c, AUTOMATIC_FSTAB_HEADER_START);
+                ace = find_line(c, AUTOMATIC_FSTAB_HEADER_END);
```

Comment:
hmm, maybe introduce both find_line() and find_line_after() so that we can do this nicely, without the strlen() stuff below.

Reply:
also, if for some reaosn the original file had start and end markers swapped weird things will happen here. hence make sure to search for the end marker only after the start marker

Reply:
Done. Searching after `acs`